src/client/dfs/dfs.c

```diff
@@ -1941,9 +1954,15 @@ dfs_lookup(dfs_t *dfs, const char *path, int flags, dfs_obj_t **_obj,
 				goto dfs_lookup_loop;
 			}
 
-			obj->value = entry.value;
+			/* Create a truncated version of the string */
+			D_STRNDUP(obj->value, entry.value, entry.value_len + 1);
+			if (obj->value == NULL)
```

Comment:
should that just be ENOMEM?

Reply:
it could be, and this code could also in theory use realloc() to truncate the existing buffer in place.  Rather than handle an additional error path here I chose the easy option of just using the longer buffer.  Not ideal as you say because it's larger but possibly preferable to failing.